crossfit/utils/torch_utils.py

```diff
@@ -99,3 +100,44 @@ def cleanup_torch_cache() -> None:
     gc.collect()
     torch.cuda.empty_cache()
     return None
+
+
+def reset_memory_tracking() -> None:
+    """
+    Resets memory counters.
+
+    This function enables memory usage statistics tracking and resets the counters
+    for peak memory usage. It handles both RMM (RAPIDS Memory Manager) and PyTorch's
+    native CUDA memory tracking, depending on the current allocator backend.
+
+    If RMM is being used as the allocator, it enables RMM statistics and pushes a new
+    statistics context. If the default PyTorch allocator is being used, it resets the
+    peak memory stats for CUDA.
+
+    Returns:
+        None
+    """
+    # TODO: This is hacky, we need to check if the allocator is rmm
+    #  and then reset the peak memory stats
+    if torch.cuda.memory.get_allocator_backend() == "pluggable":
```

Comment:
@bdice , Curious if you know of a there a better way to check this, this feels very hacky.

Reply:
I haven't used any of these before but would something like [`_get_current_allocator`](https://github.com/pytorch/pytorch/blob/22e19bd2d70409c2908edf0b0d00abb9209e3aaa/torch/cuda/memory.py#L972C16-L972C27) work? Or maybe check if RMM is already initialized with [`is_initialized`](https://github.com/rapidsai/rmm/blob/6489bb7df63a3784b4a94067e3a8fa8917523ab7/python/rmm/rmm/rmm.py#L94)?

Reply:
I tried,  but the type is still `torch.cuda`, so unsure how to check  using that

```
In [12]: type(torch.cuda.memory._get_current_allocator())
Out[12]: torch.cuda.memory._CUDAAllocator
```

Reply:
It doesn't look like Torch tracks anything except the allocate and deallocate function pointers. I don't think that is enough information to know what backend was used. You would probably need to track this in some global state when enabling the RMM allocator.

Reply:
Thanks Bradley, let me think of a better way to do this. 

Unfortunately a global state may not always work. I am pushing some dask level changes (https://github.com/rapidsai/dask-cuda/pull/1392), I can use that but dont think i can always force users to use this method .

Reply:
We dont have a good way to handle this today. Will follow up on Pytorch, resolving for now